vault-core-it/vault-core-integration-tests/src/main/java/org/apache/jackrabbit/vault/packaging/integration/ReferenceableIdentifiersImportIT.java

```diff
@@ -401,6 +422,138 @@ public void testImportDupPolicyLegacy() throws RepositoryException, IOException,
+        admin.save();
+
+        String id1 = asset.getIdentifier();
+
+        File pkgFile = exportContentPackage(srcPath);
+
+        String dstPath = srcPath + "-renamed";
+        admin.move(srcPath, dstPath);
+        assertNodeMissing(srcPath);
+
+        try {
+            installContentPackage(pkgFile, policy);
+        } catch (Exception ex) {
+            if (expectedException == null) {
+                throw ex;
+            } else {
+                assertTrue("expected: " + expectedException + ", but got: " + ex.getClass(), expectedException.isInstance(ex));
+                if (expectedRootCause != null) {
+                    Throwable rc = ExceptionUtils.getRootCause(ex);
+                    assertTrue("expected: " + expectedRootCause + ", but got: " + rc.getClass(), expectedRootCause.isInstance(rc));
+                }
+                // expected exception -> test done
+                return;
+            }
+        }
+
+        if (expectRenamedNodeKept) {
+            assertNodeExists(dstPath);
+        } else {
+            assertNodeMissing(dstPath);
+        }
+
+        assertNodeExists(srcPath);
+        assertNodeExists(srcPath + "/binary.txt");
+
+        Node asset2 = testRoot.getNode(srcName);
+        String id2 = asset2.getIdentifier();
+        if (expectNewId) {
+            assertNotEquals(id1, id2);
+        } else {
+            assertEquals(id1, id2);
+        }
+    }
+
+    private Node addFileNode(Node parent, String name) throws Exception {
+
+        ValueFactory valueFactory = parent.getSession().getValueFactory();
+        Binary contentValue = valueFactory.createBinary(new ByteArrayInputStream("Hello, world!".getBytes()));
+        Node fileNode = parent.addNode(name, NodeType.NT_FILE);
+        Node resNode = fileNode.addNode("jcr:content", "nt:resource");
+        resNode.setProperty("jcr:mimeType", "text/plain");
+        resNode.setProperty("jcr:data", contentValue);
+
+        return fileNode;
+    }
+
+    private File exportContentPackage(String path) throws Exception {
+        ExportOptions opts = new ExportOptions();
+        DefaultMetaInf inf = new DefaultMetaInf();
+        DefaultWorkspaceFilter filter = new DefaultWorkspaceFilter();
+
+        PathFilterSet pfs = new PathFilterSet(path);
+        pfs.addInclude(new DefaultPathFilter(path + "/.*"));
+        filter.add(pfs);
+
+        inf.setFilter(filter);
+        Properties props = new Properties();
+        props.setProperty(PackageProperties.NAME_GROUP, "jackrabbit/test");
+        props.setProperty(PackageProperties.NAME_NAME, "test-package");
+        inf.setProperties(props);
+
+        opts.setMetaInf(inf);
+        File pkgFile = File.createTempFile("testImportMovedResource", ".zip");
+        try (VaultPackage pkg = packMgr.assemble(admin, opts, pkgFile)) {
+            return pkg.getFile();
+        }
+    }
+
+    private void installContentPackage(File pkgFile, IdConflictPolicy policy)
```

Comment:
This is contained in the parent class `IntegrationTestBase` already.

Reply:
Do you mean "extractVaultPackage"? That seems to search for the file inside the class path; but in this case it's in a Junit temp folder...

Reply:
right, I would still recommend to refactor this method to leverage the high-level VaultPackage.extract(...) instead of using the low-level `Importer.run` only. Also I don't understand why you cannot export with right filter in `exportContentPackage()` and need to overwrite that here.

Reply:
I believe the filter statement is not needed here.

Reply:
Please see update.